src/plugins/core/borders.ts

```diff
@@ -90,9 +90,55 @@ export class BordersPlugin extends CorePlugin<BordersPluginState> implements Bor
         const elements = [...cmd.elements].sort((a, b) => b - a);
         for (const group of groupConsecutive(elements)) {
           if (cmd.dimension === "COL") {
-            this.shiftBordersHorizontally(cmd.sheetId, group[group.length - 1] + 1, -group.length);
+            if (group[0] == this.getters.getNumberCols(cmd.sheetId)) {
+              for (let row: HeaderIndex = 0; row < this.getters.getNumberRows(cmd.sheetId); row++) {
+                this.history.update(
+                  "borders",
+                  cmd.sheetId,
+                  group[0] + 1,
+                  row,
+                  "vertical",
+                  undefined
+                );
+              }
+            }
+            if (group[group.length - 1] == 0) {
```

Comment:
**always** use `===` in js. Typescript should mostly prevent issues, but `==` cast the variables before the comparison ... which lead to some interesting results. `0 == []` is true for example